code/Migration.php

```diff
@@ -14,4 +14,189 @@
 
 	abstract public function down();
 
+
+	#######################################
+	## DATABASE MIGRATION HELPER METHODS ##
+	#######################################
+
+	/**
+	 * Returns true if the table exists in the database
+	 *
+	 * @param string $table
+	 * @return boolean
+	 */
+	protected static function tableExists($table) {
+		$tables = DB::tableList();
+		return array_key_exists(strtolower($table), $tables);
+	}
+
+	/**
+	 * Returns true if a column exists in a database table
+	 *
+	 * @param string $table
+	 * @param string $column
+	 * @return boolean
+	 */
+	protected static function tableColumnExists($table, $column) {
+		if (!self::tableExists($table)) return false;
+		$columns = self::getTableColumns($table);
+		return array_key_exists($column, $columns);
+	}
+
+	/**
+	 * Returns true if an array of columns exist on a database table
+	 *
+	 * @param string $table
+	 * @param array $columns
+	 * @return boolean
+	 */
+	protected static function tableColumnsExist($table, array $columns) {
+		if (!self::tableExists($table)) return false;
+		return count(array_intersect($columns, array_keys(self::getTableColumns($table)))) === count($columns);
+	}
+
+	/**
+	 * Returns an array of columns for a database table
+	 *
+	 * @param string $table
+	 * @return array (empty if table doesn't exist)
+	 */
+	protected static function getTableColumns($table) {
+		if (!self::tableExists($table)) return [];
+		return DB::fieldList($table);
+	}
+
+	/**
+	 * Drops columns from a database table.
+	 * Returns false if the table or any of the columns do not exist.
+	 * Returns true if the SQL query was executed.
+	 *
+	 * @param string $table
+	 * @param array $columns
+	 * @return boolean
+	 */
+	protected static function dropColumnsFromTable($table, array $columns) {
+		$queried = false;
```

Comment:
Just a muse, but this is where array_intersect and array_keys is handy too. Start out getting the intersection, if it's empty, return false. It will be empty if there is no table or if the specified array of columns aren't found. Then, just iterate through that array and issue the  `ALTER` query on the (escaped) table and return true at the end (since we already know the array wasn't empty).